src/Miunie.WindowsApp/App.xaml.cs

```diff
@@ -31,8 +35,41 @@ sealed partial class App : Application
         /// </summary>
         public App()
         {
-            this.InitializeComponent();
-            this.Suspending += OnSuspending;
+            InitializeComponent();
+            Suspending += OnSuspending;
+            UnhandledException += App_UnhandledException;
+        }
+
+        private async void App_UnhandledException(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
+        {
+            e.Handled = true;
+            System.Diagnostics.Debug.WriteLine(e.Exception);
+
+            var logger = SimpleIoc.Default.GetInstance<ILogWriter>();
+            logger.LogError($"{e.Message}");
+
+            var confirmRestart = new ContentDialog
+            {
+                Title = "Opps! Wasn't expecting that...",
+                Content = "Miunie encountered an error that she was unprepared for. Would you like to reset Miunie with a clean restart?",
+                PrimaryButtonText = "Yes, restart Miunie",
+                CloseButtonText = "Try my luck and continue"
```

Comment:
While I always love the creative messages you come up with, this one is more like Windows speaking. 😅 

For that reason, I'd suggest we stick with [Fluent design writing style guidelines](https://docs.microsoft.com/en-us/windows/uwp/design/style/writing-style#error-messages) for errors.

Especially the "Be crisp and clear by eliminating extraneous information." part.

So just a simple "An unexpected error occurred, would you like to restart Miunie, or ignore this problem?"

Or at least something along those lines. 😊 
Sorry to nit-pick like this, I just feel like there is a time and place for style. 😅